apps/automation-v2/src/subgraph/subgraph-request.ts

```diff
@@ -1,20 +1,35 @@
-import axios from 'axios';
+import axios, { AxiosError } from 'axios';
 
 import { SubGraphQuery, SubGraphResponse, SubgraphError } from './types';
+import { Logger } from '@mountainpath9/overlord';
 
-export const subgraphRequest = <R extends SubGraphResponse<object>>(subgraphUrl: string, query: SubGraphQuery) => {
+export const subgraphRequest = <R extends SubGraphResponse<object>>(
+  subgraphUrl: string,
+  query: SubGraphQuery,
+  logger?: Logger
+) => {
   const subgraphRequest = async () => {
+    logger?.info(
+      `try axios post to: \n{ "subgraphUrl": "${subgraphUrl}", "subgraphQuery": ${JSON.stringify(
+        query.query
+      ) }}`
+    );
     try {
-      const { data } = await axios.post<R>(subgraphUrl, query);
-
-      if ((data?.errors || []).length > 0) {
-        const firstErrorMessage = data.errors![0].message;
-        throw new Error(firstErrorMessage);
+      const res = await axios.post<R>(subgraphUrl, query);
+      logger?.info(`res status code ${res.status} ${res.statusText}`);
+      if ((res.data?.errors || []).length > 0) {
+        const firstErrorMessage = res.data.errors![0].message;
+        throw new AxiosError(
+          firstErrorMessage,
+          res.statusText,
+          res.config,
+          res.request,
+          res
+        );
       }
-
-      return data;
+      return res;
     } catch (err) {
```

Comment:
why throw an AxiosError here, then catch it below and throw as a SubgraphError?

Reply:
src reused from https://github.com/TempleDAO/temple/blob/stage/apps/dapp/src/hooks/use-subgraph-request.tsx

Initially thought about returning an axios error, that way we could add more details to the `firstErrorMessage` instead of a single string. Regarding the `SubgraphError`, I guess the initial idea of the person who implemented it was to always throw a SubgraphError type from this function, prob I should just throw `AxiosError` from this function instead, wdyt?